File: php/offer3_arrays/Arrays.php
<?php

class Arrays {
	private $arr;
	
	public function cin($a,$nu)
	{
		$this->arr = $a;
		$i = 0;
		$j = count($this->arr[0])-1;
		
		while($i<count($this->arr)&&$j>=0){
			$u = $i;
			$v = $j;
			if($nu==$this->arr[$u][$v]){
				return true;
			}
			
			if($nu>$this->arr[$u][$v]){
				$i++;
			}
			
			if($nu<$this->arr[$u][$v]){
				$j--;
			}
		}
		return false;
	}
}

File: java/offer3_arrays/src/offer3_arrays/Arrays.java
package offer3_arrays;

public class Arrays {
	private int arr[][];
	
	public boolean cin(int a[][],int nu)
	{
		this.arr = a;
		int i = 0;
		int j = arr[0].length-1;
		
		while(i<arr.length&&j>=0){
			int u = i;
			int v = j;
			if(nu==arr[u][v]){
				return true;
			}
			
			if(nu>arr[u][v]){
				i++;
			}
			
			if(nu<arr[u][v]){
				j--;
			}
		}
		return false;
	}
}
